### ISSUE TITLE
终于生成bin文件了，感谢大佬的支持

### ISSUE THREAD

**Issue by shuidongliu-shui** [AngusMonroe/ContrastPool#10]
在大佬的指点下，项目总算是跑起来了，只是我又产生了新的问题。我生成bin文件用的是相关矩阵，因为相关矩阵的维度都是100*100，但是时间序列特征那个维度不一样，有的是200*100，有的是300*100，有的是240*100，有的是150*100，诸如此类。那这个项目没有用到时间序列特征这些文件吧，我感觉我还需要更多的学习，我再看看，再次感谢大佬的指点。
# -*- coding: utf-8 -*-
import pandas as pd
import numpy as np
import networkx as nx
import os  # To create directories
import shutil
import scipy.io
import dgl
import torch
import glob
import csv
import re
import json
from tqdm import tqdm
from dgl.data.utils import save_graphs
from sklearn.model_selection import StratifiedKFold, train_test_split

import os

def _load_matrix_subject_with_files(files, remove_negative=False):
    subjects = []
    for file in files:
        mat = scipy.io.loadmat(file)
        mat = mat["data"]
        print(f"Loaded data shape: {mat.shape}")
        np.fill_diagonal(mat, 0)
        if remove_negative:
            mat[mat < 0] = 0
        subjects.append(mat)
    return np.array(subjects)

def construct_dataset(path):

    feat_dir = path
    if not os.path.exists(feat_dir):
        raise FileNotFoundError(f"目录 {feat_dir} 不存在！")
    G_dataset = []
    Labels = []
    group2idx = {}
    paths = glob.glob(feat_dir + '/*/' + '*_schaefer100_correlation_matrix.mat', recursive=True)
    if not paths:
        raise FileNotFoundError(f"在 {feat_dir} 下未找到任何匹配 *_features_timeseries.mat 的文件！")
    feats = _load_matrix_subject_with_files(paths)

    for j in tqdm(range(len(feats))):
        name = paths[j].split('/')[-1]
        group = re.findall('sub-([^\d]+)', name)[0]
        if group not in group2idx.keys():
            group2idx[group] = len(group2idx.keys())
        i = group2idx[group]

        G = nx.DiGraph(np.ones([feats[j].shape[0], feats[j].shape[0]]))
        graph_dgl = dgl.from_networkx(G)

        graph_dgl.ndata['N_features'] = torch.from_numpy(feats[j])
        # Include edge features
        weights = []
        for u, v, w in G.edges.data('weight'):
            # if w is not None:
            weights.append(w)
        graph_dgl.edata['E_features'] = torch.Tensor(weights)

        G_dataset.append(graph_dgl)
        Labels.append(i)

    Labels = torch.LongTensor(Labels)
    graph_labels = {"glabel": Labels}
    if not os.path.exists('./bin_dataset/'):
        os.mkdir('./bin_dataset/')
    print(Labels.shape)
    print(len(G_dataset))
    save_graphs("c:\\ABIDE\\abide_schaefer100.bin", G_dataset, graph_labels)

if __name__ == '__main__':
    import traceback  # 导入 traceback 模块

    error_name = []
    path = "C:\\ABIDE"
    try:
        construct_dataset(path)
    except Exception as e:
        # 打印详细错误信息
        print(f"[ERROR]: {path} → {str(e)}")
        traceback.print_exc()  # 打印堆栈跟踪
        error_name.append(path)
    '''file_name_list = ['ABIDE']

    for data_name in file_name_list:
        try:
            construct_dataset(data_name)
        except Exception as e:
            # 打印详细错误信息
            print(f"[ERROR]: {data_name} → {str(e)}")
            traceback.print_exc()  # 打印堆栈跟踪
            error_name.append(data_name)'''

    print("错误数据集列表:", error_name)
    print('Done!')

**Comment by AngusMonroe** [comment]
Both the feature matrix and adjacency matrix come from the correlation matrix. We didn't use the time series directly for brain network construction.

**Comment by shuidongliu-shui** [comment]
原来如此，作为一名菜鸟研究生，我有了毕设的思路。如果我在时间序列矩阵上面做点功夫，添加一些参数，或许我就可以毕业了，再次感谢大佬的支持和回答。
